main/lsp/requests/completion.cc

```diff
@@ -839,8 +939,30 @@ vector<unique_ptr<CompletionItem>> CompletionTask::getCompletionItems(LSPTypeche
 
     vector<core::NameRef> similarLocals;
     if (params.enclosingMethod.exists()) {
-        auto locals = localNamesForMethod(typechecker, params.enclosingMethod, params.queryLoc);
-        similarLocals = allSimilarLocalNames(gs, locals, params.prefix);
+        // Slyly reuse `UnresolvedIdent::Kind` to both determine what kind of
+        // thing we're going to find and to determine the search space for
+        // `fieldsForClass`.
+        //
+        // TODO: for empty prefixes, we would like to complete instance/class
+        // variables along with locals.
+        //
+        // TODO: for a prefix of just "@", we should provide class variables
+        // along with instance variables.
```

Comment:
There are tests that try completing after just a `@`, but I don't think the necessary information makes its way into the AST to enable us to see that we have that prefix there.  Unless there need to be some extra gymnastics in `CompletionTask::runRequest` to look for a bare `@` prior to the completion point?

Reply:
Yes, when I looked at this it looked like the parser wasn't always good enough:

https://sorbet.run/?arg=--print=parse-tree-whitequark#%23%20typed%3A%20true%0Adef%20foo%0A%20%20puts%20'this%20is%20missing'%0A%20%20%40%0A%20%20puts%20'this%20is%20after'%0Aend

I was hoping to snipe you into writing test cases for what didn't work, and then using http://go/bad-parse for whatever doesn't work because the parser isn't yet good enough. Fixing this doesn't seem any harder than getting `x.` to parse, but we just have to put time into it.